src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -373,5 +378,131 @@ private static void TrimInaccessibleText(ArrayBuilder<SourceText> segments)
                 segments.Add(writer.ToSourceText());
             }
         }
+
+        /// <summary>
+        /// Delegates to SourceTexts within the CompositeText to determine line information.
+        /// </summary>
+        private sealed class CompositeLineInfo : TextLineCollection
+        {
+            private readonly CompositeText _compositeText;
+
+            // The starting line number for the correspondingly indexed SourceTexts in _compositeText.Segments
```

Comment:
make an actual doc comment.

Reply:
consider saying things like "will have the same length as _compositeText.Segments"